x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/expression/predicate/regex/StringPatternTests.java

```diff
@@ -12,78 +12,78 @@
 
 public class StringPatternTests extends ESTestCase {
 
-    private LikePattern like(String pattern, char escape) {
-        return new LikePattern(pattern, escape);
+    private WildcardPattern like(String pattern) {
+        return new WildcardPattern(pattern);
     }
 
     private RLikePattern rlike(String pattern) {
         return new RLikePattern(pattern);
     }
 
-    private boolean matchesAll(String pattern, char escape) {
-        return like(pattern, escape).matchesAll();
+    private boolean likeMatchesAll(String pattern) {
+        return like(pattern).matchesAll();
     }
 
-    private boolean exactMatch(String pattern, char escape) {
-        String escaped = pattern.replace(Character.toString(escape), StringUtils.EMPTY);
-        return escaped.equals(like(pattern, escape).exactMatch());
+    private boolean likeExactMatch(String pattern) {
+        String escaped = pattern.replace("\\", StringUtils.EMPTY);
+        return escaped.equals(like(pattern).exactMatch());
     }
 
-    private boolean matchesAll(String pattern) {
+    private boolean rlikeMatchesAll(String pattern) {
         return rlike(pattern).matchesAll();
     }
 
-    private boolean exactMatch(String pattern) {
+    private boolean rlikeExactMatch(String pattern) {
         return pattern.equals(rlike(pattern).exactMatch());
     }
 
-    public void testWildcardMatchAll() throws Exception {
-        assertTrue(matchesAll("%", '0'));
-        assertTrue(matchesAll("%%", '0'));
+    public void testWildcardMatchAll() {
+        assertTrue(likeMatchesAll("*"));
+        assertTrue(likeMatchesAll("*"));
```

Comment:
This looks like a typo:
```suggestion
        assertTrue(likeMatchesAll("**"));
```